File: EDA_Visualisation_GenderPayGap.py
# -*- coding: utf-8 -*-
"""
Created on Mon Mar 18 21:06:49 2024

@author: Maja
"""
#####    Data Visualization 
 
###          QUESTIONS

# 0. Import libraries

import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import matplotlib.ticker as mtick


### 1.	Import Premium and Claim data and merge both data sets into one data.
GGPG_data = pd.read_csv("C:/Users/Maja/Documents/Learning/MS_Projects/Portfolio 2024/GitHub Portfolio/4_Data Visualisation/Glassdoor Gender Pay Gap.csv")

GGPG_data.head(n=20)
print("GGPG_data shape: ", GGPG_data.shape)
GGPG_data.info()
print(list(GGPG_data))


# 2.	BasePay & Bonus by Department, Job Role, Seniority:
# 2.1 By Department: We'll visualize BasePay and Bonus across departments,

# Department by Base Pay
# Set the color palette
sns.set_palette("Paired")

# Create the bar plot
plt.figure(figsize=(10, 6))
sns.barplot(x='Dept', y='BasePay', data=GGPG_data)

# Add titles and labels
plt.title('Base Pay by Department')
plt.xlabel('Dept')
plt.ylabel('Base Pay')

# Format y-axis with commas
plt.gca().yaxis.set_major_formatter(mtick.FuncFormatter(lambda x, _: f'{int(x):,}'))

# Rotate x-axis labels
plt.xticks(rotation=45, ha='right')

# Minimal theme
sns.despine()

# Show plot
plt.show()

# Department by Bonus
# Set the color palette
sns.set_palette("Paired")

# Create the bar plot
plt.figure(figsize=(10, 6))
sns.barplot(x='Dept', y='Bonus', data=GGPG_data)

# Add titles and labels
plt.title('Bonus by Department')
plt.xlabel('Dept')
plt.ylabel('Bonus')

# Format y-axis with commas
plt.gca().yaxis.set_major_formatter(mtick.FuncFormatter(lambda x, _: f'{int(x):,}'))

# Rotate x-axis labels
plt.xticks(rotation=45, ha='right')

# Minimal theme
sns.despine()

# Show plot
plt.show()

# Box Plot: To compare BasePay across different levels of Seniority

# Set the color palette
sns.set_palette("Paired")

# Create the box plot
plt.figure(figsize=(10, 6))
sns.boxplot(x='Seniority', y='BasePay', data=GGPG_data, hue='Seniority')

# Add titles and labels
plt.title('BasePay by Seniority Level')
plt.xlabel('Seniority Level')
plt.ylabel('BasePay')

# Show plot
plt.show()

# Box Plot: To compare Bonus across different levels of Seniority
# Set the color palette
sns.set_palette("Paired")

# Create the box plot
plt.figure(figsize=(10, 6))
sns.boxplot(x='Seniority', y='Bonus', data=GGPG_data, hue='Seniority')

# Add titles and labels
plt.title('Bonus by Seniority Level')
plt.xlabel('Seniority Level')
plt.ylabel('Bonus')

# Move the legend to the top right outside the plot
plt.legend(title='Seniority', bbox_to_anchor=(1.05, 1), loc='upper left')

# Show plot
plt.tight_layout()  # Adjust layout to ensure everything fits without overlap
plt.show()

# 2.2	By Job Role: We'll then explore the BasePay and Bonus distribution across different job roles.
# Set the color palette
sns.set_palette("Paired")

# Plot 1: BasePay by JobTitle
plt.figure(figsize=(12, 6))
sns.barplot(x='JobTitle', y='BasePay', data=GGPG_data, palette='Paired')
plt.title('Base Pay by Job Title')
plt.xlabel('Job Title')
plt.ylabel('Base Pay')
plt.xticks(rotation=45)
plt.gca().yaxis.set_major_formatter(plt.FuncFormatter(lambda x, loc: "{:,}".format(int(x))))
plt.show()

# Plot 2: Bonus by JobTitle
plt.figure(figsize=(12, 6))
sns.barplot(x='JobTitle', y='Bonus', data=GGPG_data, palette='Paired')
plt.title('Bonus by Job Title')
plt.xlabel('Job Title')
plt.ylabel('Bonus')
plt.xticks(rotation=45)
plt.gca().yaxis.set_major_formatter(plt.FuncFormatter(lambda x, loc: "{:,}".format(int(x))))
plt.show()

# 3.	Mean BasePay Visualization:
# 3.1	Bar Chart by Department: Calculate the mean BasePay for each department and visualize it with a bar chart, using a palette from R's Color Brewer.
# Calculate mean Base Pay per Department

# Calculate mean Base Pay per Department
mean_BasePay_per_Dept = GGPG_data.groupby('Dept')['BasePay'].mean().reset_index()

# Rename columns for clarity 
mean_BasePay_per_Dept.columns = ['Dept', 'BasePay_avg']

# Set the color palette
sns.set_palette("Paired")

# Plot the bar chart
plt.figure(figsize=(10, 6))
sns.barplot(x='Dept', y='BasePay_avg', data=mean_BasePay_per_Dept, palette="Paired")

# Add titles and labels
plt.title('Average Base Pay by Department')
plt.xlabel('Department')
plt.ylabel('Mean of Base Pay')

# Rotate the x-axis labels for better readability
plt.xticks(rotation=45, ha='right')

# Apply minimal theme
sns.despine()

# Show plot
plt.tight_layout()  # Adjust the layout to fit everything
plt.show()


# Bonus by Department: Visualize the average Bonus across departments.
# Calculate mean Bonus per Department

mean_Bonus_per_Dept = GGPG_data.groupby('Dept')['Bonus'].mean().reset_index()

# Rename columns for clarity 
mean_Bonus_per_Dept.columns = ['Dept', 'Bonus_avg']

# Set the color palette
sns.set_palette("Paired")

# Plot the bar chart
plt.figure(figsize=(10, 6))
sns.barplot(x='Dept', y='Bonus_avg', data=mean_Bonus_per_Dept, palette="Paired")

# Add titles and labels
plt.title('Average Bonus by Department')
plt.xlabel('Department')
plt.ylabel('Mean of Bonus')

# Rotate the x-axis labels for better readability
plt.xticks(rotation=45, ha='right')

# Apply minimal theme
sns.despine()

# Show plot
plt.tight_layout()  # Adjust the layout to fit everything
plt.show()

# 3.2 BasePay & Bonus by Department: Visualize the average BasePay and Bonus across departments.

# Step 1: Merge the two datasets
combined_data = pd.merge(mean_BasePay_per_Dept, mean_Bonus_per_Dept, on='Dept')

# Step 2: Pivot the merged dataset from wide to long format
combined_data = combined_data.melt(id_vars='Dept', 
                                   value_vars=['BasePay_avg', 'Bonus_avg'], 
                                   var_name='Pay_Type', 
                                   value_name='Average')

# Step 3: Plot the bar chart
plt.figure(figsize=(10, 6))
sns.barplot(x='Dept', y='Average', hue='Pay_Type', data=combined_data, palette="Paired")

# Add titles and labels
plt.title('Average Base Pay and Bonus by Department')
plt.xlabel('Department')
plt.ylabel('Average Pay')

# Rotate the x-axis labels for better readability
plt.xticks(rotation=45, ha='right')

# Apply minimal theme
sns.despine()
# Show plot
plt.tight_layout() 
plt.show()

# 3.3	BasePay & Bonus by Job Role: Similarly, visualize the average BasePay and Bonus by job roles.
# Mean of Base Pay & Bonus by Job Role
#  Calculate mean BasePay and Bonus per Job Title
mean_JobRole = GGPG_data.groupby('JobTitle').agg(
    avg_BasePay=('BasePay', 'mean'),
    avg_Bonus=('Bonus', 'mean')
).reset_index()

#  Pivot the DataFrame from wide to long format
mean_JobRole_long = mean_JobRole.melt(
    id_vars='JobTitle', 
    value_vars=['avg_BasePay', 'avg_Bonus'], 
    var_name='Pay_Type2', 
    value_name='Average2'
)

#  Plotting the bar chart
plt.figure(figsize=(12, 6))
sns.barplot(x='JobTitle', y='Average2', hue='Pay_Type2', data=mean_JobRole_long, palette='Paired')

#  Customize the plot
plt.title('Average Base Pay and Bonus by Job Title')
plt.xlabel('Job Title')
plt.ylabel('Average Pay')
plt.xticks(rotation=45, ha='right')  # Rotate x-axis labels for better readability
sns.despine()  # Apply minimal theme

# Adjust layout to ensure everything fits
plt.tight_layout()

# Show the plot
plt.show()

# 4.	Gender-Based Pay Visualizations:
# 4.1	BasePay by Gender: Use boxplots to compare the distributions of BasePay across genders.

# Set the color palette
sns.set_palette("Paired")

# Create the boxplot
plt.figure(figsize=(10, 6))
sns.boxplot(x='Gender', y='BasePay', data=GGPG_data, palette='Paired')

# Add titles and labels
plt.title('BasePay Distribution by Gender')
plt.xlabel('Gender')
plt.ylabel('Base Pay')

# Apply minimal theme
sns.despine()  # Removes the top and right spines for a cleaner look

# Show the plot
plt.show()

# 4.2	Bonus by Gender: Visualize the distribution of Bonus amounts across genders using boxplots.
# Set the color palette
sns.set_palette("Paired")

# Create the boxplot
plt.figure(figsize=(10, 6))
sns.boxplot(x='Gender', y='Bonus', data=GGPG_data, palette='Paired')

# Add titles and labels
plt.title('Bonus Distribution by Gender')
plt.xlabel('Gender')
plt.ylabel('Bonus')

# Apply minimal theme
sns.despine()  # Removes the top and right spines for a cleaner look

# Show the plot
plt.show()
# 4.3	Scatter Plot: Explore the relationship between BasePay and Bonus, with points coloured by gender.

# Create the plot
plt.figure(figsize=(10, 6))

# Create the scatter plot
plot = sns.scatterplot(data=GGPG_data, x='BasePay', y='Bonus', hue='Gender', alpha=0.7, palette='Paired')

# Add titles and labels
plot.set_title('Scatter Plot of BasePay vs. Bonus by Gender')
plot.set_xlabel('Base Pay')
plot.set_ylabel('Bonus')

# Show the legend
plt.legend(title='Gender')

# Apply a minimal theme
plt.style.use('seaborn-darkgrid')

# Display the plot
plt.show()

# 4.4	Facet Grid: View separate scatter plots for each gender, offering a clear comparison.
# Set the overall aesthetics
sns.set(style="whitegrid")

# Create the scatter plot with facet grid
g = sns.FacetGrid(GGPG_data, col="Gender", height=5, aspect=1)
g.map(sns.scatterplot, "BasePay", "Bonus", alpha=0.7)

# Add titles and labels
g.set_axis_labels("Base Pay", "Bonus")
g.set_titles("Gender: {col_name}")
g.fig.suptitle("BasePay vs. Bonus Faceted by Gender", y=1.05, fontsize=16)

# Adjust layout to prevent overlap
plt.tight_layout()

# Show plot
plt.show()

# 4.5 Violin Plot: Combine the distributions of BasePay and Bonus into a single plot, visualizing differences by gender.
# Melting the data
melted_data = GGPG_data.melt(id_vars='Gender', value_vars=['BasePay', 'Bonus'], var_name='variable', value_name='value')

# Create the figure and subplots
fig, axes = plt.subplots(nrows=1, ncols=2, figsize=(14, 6), sharey=False)

# Plot for BasePay
sns.violinplot(data=melted_data[melted_data['variable'] == 'BasePay'], x='Gender', y='value', hue='Gender', palette='Paired', ax=axes[0], trim=False)
axes[0].set_title('Distribution of BasePay by Gender')
axes[0].set_xlabel('Gender')
axes[0].set_ylabel('Value')

# Plot for Bonus
sns.violinplot(data=melted_data[melted_data['variable'] == 'Bonus'], x='Gender', y='value', hue='Gender', palette='Paired', ax=axes[1], trim=False)
axes[1].set_title('Distribution of Bonus by Gender')
axes[1].set_xlabel('Gender')
axes[1].set_ylabel('Value')

# Adjust layout
plt.tight_layout()

# Show the plot
plt.show()

# 4.6	Bar Plot: Compare the average BasePay and Bonus between genders using a bar plot.

# Compute average BasePay and Bonus by Gender
mean_pay_bonus = GGPG_data.groupby('Gender').agg(
    BasePay_avg=('BasePay', 'mean'),
    Bonus_avg=('Bonus', 'mean')
).reset_index()

# Melt the data for plotting
mean_pay_bonus_melted = pd.melt(mean_pay_bonus, id_vars='Gender', value_vars=['BasePay_avg', 'Bonus_avg'], var_name='Variable', value_name='Average Value')

# Create the grouped bar plot
plt.figure(figsize=(10, 6))
plot = sns.barplot(data=mean_pay_bonus_melted, x='Gender', y='Average Value', hue='Variable', palette='Paired')

# Add titles and labels
plot.set_title('Average BasePay and Bonus by Gender')
plot.set_xlabel('Gender')
plot.set_ylabel('Average Value')
plot.legend(title='Variable')

# Apply a minimal theme
sns.set_style('whitegrid')

# Show the plot
plt.show()

# 5.	Heat Map Visualizations:
# 5.1	By Job Title and Department: Create a heat map showing the average BasePay by Job Title and Department.
# Compute average BasePay by Dept and JobTitle

# Compute average BasePay by Dept and JobTitle
avgBasePay_Dept_JobTitle = GGPG_data.groupby(['Dept', 'JobTitle']).agg(
    avg_BasePay=('BasePay', 'mean')
).reset_index()

# Pivot the data for heatmap
heatmap_data = avgBasePay_Dept_JobTitle.pivot(index='JobTitle', columns='Dept', values='avg_BasePay')

# Create the heatmap
plt.figure(figsize=(10, 8))
heatmap = sns.heatmap(
    heatmap_data,
    cmap='coolwarm',  # Use a gradient from cool to warm colors
    annot=True,       # Show values in the heatmap
    fmt=".1f",        # Format the values
    linewidths=0.5,   # Line width between cells
    cbar_kws={'label': 'Average Base Pay'}  # Color bar label
)

# Set titles and labels
plt.title('Heatmap of Job Title and Department with Average Base Pay')
plt.xlabel('Dept')
plt.ylabel('Job Title')

# Rotate x-axis labels for better readability
plt.xticks(rotation=45, ha='right')

# Show the plot
plt.tight_layout()
plt.show()

# 5.2	Count of Job Titles by Department: Visualize the count of job titles across departments using a heat map.
# Compute count of JobTitle by Dept
JobTitle_count = GGPG_data.groupby(['Dept', 'JobTitle']).size().reset_index(name='JT_count')

# Pivot the data for heatmap
heatmap_data = JobTitle_count.pivot(index='JobTitle', columns='Dept', values='JT_count')

# Create the heatmap
plt.figure(figsize=(10, 8))
heatmap = sns.heatmap(
    heatmap_data,
    cmap='YlGnBu',  # Use a gradient from light to dark green
    annot=True,     # Show counts in the heatmap
    fmt='d',        # Format for integer values
    linewidths=0.5, # Line width between cells
    cbar_kws={'label': 'Count'}  # Color bar label
)

# Set titles and labels
plt.title('Heat Map of Job Title Counts by Department')
plt.xlabel('Department')
plt.ylabel('Job Title')

# Rotate x-axis labels for better readability
plt.xticks(rotation=45, ha='right')

# Show the plot
plt.tight_layout()
plt.show()

# 5.3	BasePay by Job Title and Gender: Explore the relationship between BasePay, job titles, and gender with a heat map.
# Compute average BasePay by JobTitle and Gender
heatmap_data = GGPG_data.groupby(['JobTitle', 'Gender']).agg(
    BasePay_avg=('BasePay', 'mean')
).reset_index()

# Pivot the data for heatmap
heatmap_pivot = heatmap_data.pivot(index='Gender', columns='JobTitle', values='BasePay_avg')

# Create the heatmap
plt.figure(figsize=(12, 8))
heatmap = sns.heatmap(
    heatmap_pivot,
    cmap='coolwarm',       # Use a gradient color map from yellow to turquoise
    annot=True,            # Show values in the heatmap
    fmt=".1f",            # Format the values
    linewidths=0.5,       # Line width between cells
    cbar_kws={'label': 'Average Base Pay'}  # Color bar label
)

# Set titles and labels
plt.title('Heatmap of Job Title and Gender with Average Base Pay')
plt.xlabel('Job Title')
plt.ylabel('Gender')

# Rotate x-axis labels for better readability
plt.xticks(rotation=45, ha='right')

# Apply a minimal theme
sns.set_style('whitegrid')

# Show the plot
plt.tight_layout()
plt.show()

# 6.	Age Distribution:
# 6.1	Histogram: Obtain a histogram to visualize the distribution of employees' ages.
# Code Example Using matplotlib:
# Create the histogram
plt.figure(figsize=(10, 6))
plt.hist(GGPG_data['Age'], bins=range(20, 100, 5), color='skyblue', edgecolor='black')

# Add titles and labels
plt.title('Distribution of Employee Ages')
plt.xlabel('Age')
plt.ylabel('Count')

# Show the plot
plt.tight_layout()
plt.show()


# Code Example Using seaborn:
# Create the histogram
plt.figure(figsize=(10, 6))
sns.histplot(GGPG_data['Age'], bins=range(20, 100, 5), color='skyblue', edgecolor='black')

# Add titles and labels
plt.title('Distribution of Employee Ages')
plt.xlabel('Age')
plt.ylabel('Count')

# Show the plot
plt.tight_layout()
plt.show()

# 7.	Gender Distribution:
# 7.1	Bar Chart: Display the count of each gender within the dataset.

# Code Example Using matplotlib:
# Count occurrences of each gender
gender_counts = GGPG_data['Gender'].value_counts()

# Create the bar plot
plt.figure(figsize=(8, 6))
plt.bar(gender_counts.index, gender_counts.values, color='lightgreen', edgecolor='black')

# Add titles and labels
plt.title('Gender Distribution')
plt.xlabel('Gender')
plt.ylabel('Count')

# Show the plot
plt.tight_layout()
plt.show()


# Code Example Using seaborn:
# Create a count plot (bar plot of counts)
plt.figure(figsize=(8, 6))
sns.countplot(data=GGPG_data, x='Gender', color='lightgreen', edgecolor='black')

# Add titles and labels
plt.title('Gender Distribution')
plt.xlabel('Gender')
plt.ylabel('Count')

# Show the plot
plt.tight_layout()
plt.show()

# 8.	Education Level Distribution:
# 8.1	Bar Chart: Visualize the count of employees at different education levels using a bar chart.
# Code Example Using matplotlib:
# Count occurrences of each education level
education_counts = GGPG_data['Education'].value_counts()

# Create the bar plot
plt.figure(figsize=(8, 6))
plt.bar(education_counts.index, education_counts.values, color='purple', edgecolor='black')

# Add titles and labels
plt.title('Distribution of Education Levels')
plt.xlabel('Education Level')
plt.ylabel('Count')

# Show the plot
plt.tight_layout()
plt.show()
    
# Code Example Using seaborn:
# Create a count plot (similar to a bar plot of counts)
plt.figure(figsize=(8, 6))
sns.countplot(data=GGPG_data, x='Education', palette='Purples', edgecolor='black')

# Add titles and labels
plt.title('Distribution of Education Levels')
plt.xlabel('Education Level')
plt.ylabel('Count')

# Show the plot
plt.tight_layout()
plt.show()

File: EDA_Visualisation_GenderPayGap.R

####  Data Visualization 

#### Background:
#### The data for analysis is a Glassdoor Gender Pay Gap in which base pay information
#### is provided for each employee for all the roles and departments.

####  QUESTIONS

## 0. Importing packages and libraries:
library(dplyr)
library(ggplot2)
library(RColorBrewer)
library(tidyr)

# 1.	Import the Data: Begin by importing the Glassdoor Gender Pay Gap data.
GGPG_data<-read.csv(file.choose(), header=T)

head(GGPG_data)
dim(GGPG_data)
summary(GGPG_data)

# 2.	BasePay & Bonus by Department, Job Role, Seniority:
# 2.1 By Department: We'll visualize BasePay and Bonus across departments,
#     utilizing a color palette from R’s Color Brewer.

ggplot(GGPG_data, aes(Dept, BasePay, fill = Dept))+
  geom_col()+
  scale_fill_brewer(palette = "Paired") +  
  labs(x = "Dept", y = " Base Pay", title = " Base Pay by Department")+
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(GGPG_data, aes(Dept, Bonus, fill = Dept))+
  geom_col()+
  scale_fill_brewer(palette = "Paired") +  
  labs(x = "Dept", y = " Bonus", title = " Bonus by Department")+
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# Box Plot: To compare BasePay across different levels of Seniority
ggplot(GGPG_data, aes(x = factor(Seniority), y = BasePay, fill = factor(Seniority))) +
  geom_boxplot() +
  labs(title = "BasePay by Seniority Level", x = "Seniority Level", y = "BasePay")

# Box Plot: To compare Bonus across different levels of Seniority
ggplot(GGPG_data, aes(x = factor(Seniority), y = Bonus, fill = factor(Seniority))) +
  geom_boxplot() +
  labs(title = "BasePay by Seniority Level", x = "Seniority Level", y = "BasePay")


# 2.2	By Job Role: We'll then explore the BasePay and Bonus distribution across different job roles.
ggplot(GGPG_data, aes(JobTitle, BasePay, fill = JobTitle))+
  geom_col()+
  scale_fill_brewer(palette = "Paired") +  
  labs(x = "JobTitle", y = " Base Pay", title = " Base Pay by JobTitlet")+
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

ggplot(GGPG_data, aes(JobTitle, Bonus, fill = JobTitle))+
  geom_col()+
  scale_fill_brewer(palette = "Paired") +  
  labs(x = "JobTitle", y = " Bonus", title = " Bonus by JobTitle")+
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

# 3.	Mean BasePay Visualization:
# 3.1	Bar Chart by Department: Calculate the mean BasePay for each department and visualize it with a bar chart, using a palette from R's Color Brewer.
# Calculate mean Base Pay per Department
mean_BasePay_per_Dept<-GGPG_data%>%
  group_by(Dept)%>%
  summarise(BasePay_avg=mean(BasePay))
mean_BasePay_per_Dept
# checking all colour pallets in the package
dev.off() 
display.brewer.all(colorblindFriendly = TRUE) 
# Plot bar chart of mean Base Pay per Department
ggplot(mean_BasePay_per_Dept, aes(x = Dept, y = BasePay_avg, fill = Dept)) +
  geom_bar(stat = "identity") +
  scale_fill_brewer(palette = "Paired") +  
  labs(x = "Dept", y = "Mean of Base Pay", title = "Average Base Pay by Department")+
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))


# 3.2 BasePay & Bonus by Department: Visualize the average BasePay and Bonus across departments.
# Calculate mean Bonus per Department
mean_Bonus_per_Dept<-GGPG_data%>%
  group_by(Dept)%>%
  summarise(Bonus_avg=mean(Bonus))
mean_Bonus_per_Dept
# Combine the mean base pay and mean bonus datasets into one, pivoting longer
combined_data <- mean_BasePay_per_Dept %>%
  inner_join(mean_Bonus_per_Dept, by = "Dept") %>%
  pivot_longer(cols = c("BasePay_avg", "Bonus_avg"),
               names_to = "Pay_Type",
               values_to = "Average")
head(combined_data)
# 2b ) Base Pay and Bonus by Department
ggplot(combined_data, aes(x = Dept, y = Average, fill = Pay_Type)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_fill_brewer(palette = "Paired") +
  labs(x = "Department", y = "Average Pay", 
       title = "Average Base Pay and Bonus by Department") +
  theme_minimal()

# 3.3	BasePay & Bonus by Job Role: Similarly, visualize the average BasePay and Bonus by job roles.
# Mean of Base Pay & Bonus by Job Role
mean_JobRole <- GGPG_data %>%
  group_by(JobTitle) %>%
  summarise(avg_BasePay = mean(BasePay),
            avg_Bonus = mean(Bonus))

mean_JobRole_long <- mean_JobRole %>%
  pivot_longer(cols = c("avg_BasePay", "avg_Bonus"),
               names_to = "Pay_Type2",
               values_to = "Average2")
mean_JobRole
mean_JobRole_long

ggplot(mean_JobRole_long, aes(x = JobTitle, y = Average2, fill = Pay_Type2 ))+
  geom_bar(stat = "identity", position = "dodge")+
  scale_fill_brewer(palette = "Paired") +
  labs(x = "Job Title", y = "Average Pay", 
       title = "Average Base Pay and Bonus by Job Title") +
  theme_minimal()+
  theme(axis.text.x = element_text(angle = 45, hjust = 1))
# 4.	Gender-Based Pay Visualizations:
# 4.1	BasePay by Gender: Use boxplots to compare the distributions of BasePay across genders.
ggplot(GGPG_data, aes(x = Gender, y = BasePay, fill = Gender)) +
  geom_boxplot() +
  scale_fill_brewer(palette = "Paired") +
  labs(title = "BasePay Distribution by Gender",
       x = "Gender",
       y = "Base Pay") +
  theme_minimal()
# 4.2	Bonus by Gender: Visualize the distribution of Bonus amounts across genders using boxplots.
ggplot(GGPG_data, aes(x = Gender, y = Bonus, fill = Gender)) +
  geom_boxplot() +
  scale_fill_brewer(palette = "Paired") +
  labs(title = "Bonus Distribution by Gender",
       x = "Gender",
       y = "Bonus") +
  theme_minimal()

# 4.3	Scatter Plot: Explore the relationship between BasePay and Bonus, with points colored by gender.
ggplot(GGPG_data, aes(x = BasePay, y = Bonus, color = Gender)) +
  geom_point(alpha = 0.7) +
  scale_color_brewer(palette = "Paired") +
  labs(title = "Scatter Plot of BasePay vs. Bonus by Gender",
       x = "Base Pay",
       y = "Bonus") +
  theme_minimal()

# 4.4	Facet Grid: View separate scatter plots for each gender, offering a clear comparison.
ggplot(GGPG_data, aes(x = BasePay, y = Bonus)) +
  geom_point(alpha = 0.7) +
  facet_wrap(~Gender) +
  labs(title = "BasePay vs. Bonus Faceted by Gender",
       x = "Base Pay",
       y = "Bonus") +
  theme_minimal()

# 4.5 Violin Plot: Combine the distributions of BasePay and Bonus into a single plot, visualizing differences by gender.
library(reshape2)
# Melting the data to combine BasePay and Bonus into one variable
melted_data <- melt(GGPG_data, id.vars = "Gender", measure.vars = c("BasePay", "Bonus"))

# Violin plot for BasePay and Bonus by Gender
ggplot(melted_data, aes(x = Gender, y = value, fill = Gender)) +
  geom_violin(trim = FALSE) +
  facet_wrap(~variable, scales = "free_y") +
  scale_fill_brewer(palette = "Paired") +
  labs(title = "Distribution of BasePay and Bonus by Gender",
       x = "Gender",
       y = "Value") +
  theme_minimal()

# 4.6	Bar Plot: Compare the average BasePay and Bonus between genders using a bar plot.
mean_pay_bonus <- GGPG_data %>%
  group_by(Gender) %>%
  summarise(BasePay_avg = mean(BasePay),
            Bonus_avg = mean(Bonus))

# Melt the data for ggplot
mean_pay_bonus_melted <- melt(mean_pay_bonus, id.vars = "Gender")

# Grouped bar plot
ggplot(mean_pay_bonus_melted, aes(x = Gender, y = value, fill = variable)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_fill_brewer(palette = "Paired") +
  labs(title = "Average BasePay and Bonus by Gender",
       x = "Gender",
       y = "Average Value",
       fill = "Variable") +
  theme_minimal()

# 5.	Heat Map Visualizations:
# 5.1	By Job Title and Department: Create a heat map showing the average BasePay by Job Title and Department.
avgBasePay_Dept_JobTitle<-GGPG_data %>%
  group_by(Dept, JobTitle)%>%
  summarise(avg_BasePay = mean(BasePay))
avgBasePay_Dept_JobTitle
# Creating the heat map of Average BasePay by JobTitle and Dept:           
ggplot(avgBasePay_Dept_JobTitle, aes(x = Dept, y = JobTitle, fill = avg_BasePay)) +
  geom_tile(color = "white") +
  labs(x = "Dept", y = "Job Title", title = "Heatmap of Job Title and Department with Average Base Pay") +
  scale_fill_gradientn(colors = c("orange", "lightyellow", "steelblue")) +  # Three-color gradient
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))

### --> This heat map would show the distribution of average BasePay or Bonus across different 
## job titles within each department. It could help you identify which departments have higher
## or lower pay for specific job titles.

# 5.2	Count of Job Titles by Department: Visualize the count of job titles across departments using a heat map.
JobTitle_count <- GGPG_data %>%
  group_by(Dept, JobTitle) %>%
  summarise(JT_count = length(JobTitle))
JobTitle_count
ggplot(JobTitle_count, aes(x = JobTitle, y = Dept, fill = JT_count)) +
  geom_tile(color = "white") +  
  geom_text(aes(label = JT_count), size = 4, color = "black") + 
  scale_fill_gradient(low = "lightgreen", high = "darkgreen") +  
  labs(x = "Job Title", y = "Department", fill = "Count",
       title = "Heat Map of Job Title Counts by Department") +
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  

# 5.3	BasePay by Job Title and Gender: Explore the relationship between BasePay, job titles, and gender with a heat map.
ggplot(GGPG_data, aes(x = JobTitle, y = Gender, fill = BasePay)) +
  geom_tile(color = "white") +
  labs(x = "Job Title", y = "Dept", title = "Heatmap of Job Title and Department with Job Titles Count") +
  scale_fill_gradient(low = "yellow2", high = "turquoise4")+
  theme_minimal() +
  theme(axis.text.x = element_text(angle = 45, hjust = 1))  

# 6.	Age Distribution:
# 6.1	Histogram: Obtain a histogram to visualize the distribution of employees' ages.
ggplot(GGPG_data, aes(x = Age)) +
  geom_histogram(binwidth = 5, fill = "skyblue", color = "black") +
  labs(title = "Distribution of Employee Ages", x = "Age", y = "Count")

# 7.	Gender Distribution:
# 7.1	Bar Chart: Display the count of each gender within the dataset.
ggplot(GGPG_data, aes(x = Gender)) +
  geom_bar(fill = "lightgreen", color = "black") +
  labs(title = "Gender Distribution", x = "Gender", y = "Count")
# 8.	Education Level Distribution:
# 8.1	Bar Chart: Visualize the count of employees at different education levels using a bar chart.
ggplot(GGPG_data, aes(x = Education)) +
  geom_bar(fill = "purple4", color = "black") +
  labs(title = "Distribution of Education Levels", x = "Education Level", y = "Count")



                                                                                                                                                                                        